sdk/logs/src/main/java/io/opentelemetry/sdk/logs/SdkLogRecordBuilder.java

```diff
@@ -119,20 +121,24 @@ public void emit() {
         this.observedTimestampEpochNanos == 0
             ? this.loggerSharedState.getClock().now()
             : this.observedTimestampEpochNanos;
-    loggerSharedState
-        .getLogRecordProcessor()
-        .onEmit(
-            context,
-            SdkReadWriteLogRecord.create(
-                loggerSharedState.getLogLimits(),
-                loggerSharedState.getResource(),
-                instrumentationScopeInfo,
-                timestampEpochNanos,
-                observedTimestampEpochNanos,
-                Span.fromContext(context).getSpanContext(),
-                severity,
-                severityText,
-                body,
-                attributes));
+    SpanContext spanContext = Span.fromContext(context).getSpanContext();
```

Comment:
This whole thing needs to be configurable.

For an approach consistent with tracing (which I think is a good baseline to consider), we need a new `LogSampler` interface analagous to [Sampler](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/Sampler.java).

The key `should` sample method might have a contract like:

```
boolean shouldSample(
      Context context,
      Severity severity,
      Value<?> body,
      Attributes attributes);
```

Notes:
- Pass individual components of LogRecord instead of `SdkReadWriteLogRecord` so we can avoid allocations if `shouldSample` returns false. Also for consistency with tracing.
- Omit resource, scope arguments for consistency with tracing
- Omit timestamp, observed timesamp arguments for consistency with tracing
- Omit severityText because it mostly duplicates severity
- Return a boolean instead of the `SamplingResult` returned by tracing because log sampling is less complicated that trace sampling

With an interface like this, we could have built in samplers for common use cases. I.e. we could have a "trace based" sampler that retains logs if the span in context is sampled.